src/svcutilcore/src/Microsoft/Tools/ServiceModel/SvcUtil/Options.cs

```diff
@@ -404,12 +403,67 @@ private void AddMscorlib(Dictionary<string, Type> foundCollectionTypes)
                 }
             }
 
-            private void AddServiceModelLib(Dictionary<string, Type> foundCollectionTypes)
+            private void LoadSMReferenceAssembly()
             {
-                Assembly serviceModelLib = typeof(ChannelFactory).Assembly;
-                if (!_parent._referencedAssemblies.Contains(serviceModelLib))
+                IList<string> referencedAssembliesArgs = _arguments.GetArguments(Options.Cmd.SMReference);
+                if (referencedAssembliesArgs != null && referencedAssembliesArgs.Count > 0)
                 {
-                    AddReferencedTypesFromAssembly(serviceModelLib, foundCollectionTypes);
+                    string smassembly = referencedAssembliesArgs[0];
+                    if(smassembly.LastIndexOf("System.ServiceModel.Primitives") == -1)
+                    {
+                        ToolConsole.WriteError("Need pass the right path of System.ServiceModel.Primitives for smreference parameter");
+                        throw new ArgumentException("Invalid smreference value");
+                    }
+
+                    string smpassembly = smassembly.Replace("System.ServiceModel.Primitives", "System.Private.ServiceModel");
+                    //for some lowercase path
+                    smpassembly = smpassembly.Replace("system.servicemodel.primitives", "system.private.servicemodel");
+
```

Comment:
This second replace should using the overload which takes StringComparison and specify IgnoreCase. Also what it's being replaced with should be the correct casing.